### DISCUSSION TITLE
Self Update FAILED because the downloaded agent FAILED hash check

### DISCUSSION THREAD

**Discussion by AnyFoever** [Ylianst/MeshCentral#6385] [Issues]
I am a Chinese user and my English is not very good. Please forgive me!Last week, I updated my MeshCentral from v0.9.84 to the latest v1.1.29.After the update was completed, about a hundred nodes were prompted with error messages:"Self Update FAILED because the downloaded agent FAILED hash check (1), URL: https://192.168.251.21:443/meshagents?id=3".
![image](https://github.com/user-attachments/assets/3f503ba2-1754-4cf9-baa8-c0edf1f807c5)
![image](https://github.com/user-attachments/assets/644b74ac-69b2-4a98-90e1-4dfcce5c4393)
---
The "Desktop" tab of the machine that prompted is missing and displays "MeshCore Recovery"
![image](https://github.com/user-attachments/assets/90a03a8f-f1fe-47f1-aff2-77ccd26a98da)
---
After checking many issues on GitHub, I found that the URL in the error message“ https://192.168.251.21:443/meshagents?id=3 "Unable to access.
![image](https://github.com/user-attachments/assets/7b720e37-7f60-412f-b227-71ae4f8f235b)
---
The meshagents page is also inaccessible
![image](https://github.com/user-attachments/assets/2c8ece52-6088-4b8c-9a9e-391bce64e3f9)
---
Later it was discovered that the reason was that I had added the following information in the configuration file:
![image](https://github.com/user-attachments/assets/c16290f5-bb37-4b96-beb3-8eb2b1430921)
---
This caused my login page to become ''https://192.168.251.21/login/"
When I added '/login/' to the 'https://192.168.251.21:443/meshagents?id=3' and 'https://192.168.251.21:443/meshagents' links, they were both accessible normally.
like this：
![image](https://github.com/user-attachments/assets/50368e9a-1d24-41cf-9633-73f9141c17c3)
---
So I batch executed 'agentupdateex  https://192.168.251.21:443/login/meshagents?id=3' in the collective instruction ，Solved the above problem.
![image](https://github.com/user-attachments/assets/072fee9c-3b87-45b3-ba71-f245f340f2a7)
---
My question is, how to make the download path automatically updated by the agent also include the "/login/"？

**Comment by si458** [comment]
The `login` part is a domain (multi-tenant) and used for separate customers, for example

All ur domains things should really be inside the `""` blank domains, unless u have a reason for not using the blank domain option?

Also, I've never heard of the command `agentupdateex` ? 

Only the `agentupdate` command I'm aware of which doesn't need any arguments and should update the agent for you

Also the could be a bug in the code? 
As indeed the self update code isn't using the correct url if u are using the `login` domain part

**Reply by AnyFoever** [reply]
Thank you for your reply！
---
My MeshCentral need a unified homepage, so the blank domains is set to another page.
like this:
![image](https://github.com/user-attachments/assets/ed29a52f-c38b-4d4b-8e5f-c4f1551534bf)
So is there any good way to solve my problem?
---
in addition，
There really is this command：
![image](https://github.com/user-attachments/assets/952d19cc-332c-46d6-a09d-fdf4f75744f1)
---
In the help information, this command can specify a path. It was all thanks to this command that my troubles were solved.
![image](https://github.com/user-attachments/assets/57263c50-1a33-46eb-9a78-4e2de80390ef)

---
This agent version is：
![image](https://github.com/user-attachments/assets/5f93e7c8-56a3-4169-a679-ff55c6065c22)
---
But, the upgraded agent does not have this command.
![image](https://github.com/user-attachments/assets/8a3a8d61-b56f-4742-abff-5e7b1fd30ed0)

**Reply by si458** [reply]
You shouldn't be using the agentupdateex as I've never seen this command and have no idea what it use to do.
So just use agentupdate instead
